npm_and_yarn/spec/dependabot/npm_and_yarn/file_updater/npm_lockfile_updater_spec.rb

```diff
@@ -168,6 +168,81 @@
           to include("19c4dba3bfce7574e28f1df2138d47ab4cc665b3")
       end
     end
+
+    context "when the packages name needs sanitizing" do
+      let(:files) { project_dependency_files("npm7/simple") }
+
+      it "restores the packages name attribute" do
+        parsed_lockfile = JSON.parse(updated_npm_lock_content)
+        expected_updated_npm_lock_content = fixture("updated_projects", "npm7", "simple", "package-lock.json")
+        expect(updated_npm_lock_content).to eq(expected_updated_npm_lock_content)
```

Comment:
Added a bunch of "updated" fixtures to make sure we don't mess up whitespace etc. Could be cool to automate this at some point like jest snapshot testing https://jestjs.io/docs/en/snapshot-testing